### PR TITLE
Adding performance tests between V1 and V2

### PR DESCRIPTION
**Pull request by Paultagoras**
## Checklist
Delete items not relevant to your PR:
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

**Review by mzitnik — COMMENTED** [review]
Can you provide an explanation how to run it

**Comment by mzitnik** [comment]
Can we also provide some early results just to understand how the output looks like

**Comment by chernser** [comment]
The medium size query looks suspiciously too good 

```
V1 
ClientBenchmark.queryBenchmarkMedium:p0.00 sample 1602.224 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.50 sample 1701.839 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.90 sample 1816.763 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.95 sample 1863.215 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.99 sample 1864.368 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.999 sample 1864.368 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.9999 sample 1864.368 ms/op

V2
ClientBenchmark.queryBenchmarkMedium:p0.00 sample 118.882 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.50 sample 146.538 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.90 sample 154.927 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.95 sample 156.500 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.99 sample 159.908 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.999 sample 160.694 ms/op
ClientBenchmark.queryBenchmarkMedium:p0.9999 sample 160.694 ms/op
```

**Comment by chernser** [comment]
The insert too 

```
Insert 

v1
ClientBenchmark.insertRawDataBenchmark:p0.00 sample 28.934 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.50 sample 34.013 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.90 sample 39.059 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.95 sample 42.880 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.99 sample 59.932 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.999 sample 88.595 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.9999 sample 101.057 ms/op
ClientBenchmark.insertRawDataBenchmark:p1.00 sample 101.057 ms/op

v2
ClientBenchmark.insertRawDataBenchmark:p0.00 sample 5.915 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.50 sample 7.700 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.90 sample 9.224 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.95 sample 10.076 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.99 sample 31.977 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.999 sample 53.668 ms/op
ClientBenchmark.insertRawDataBenchmark:p0.9999 sample 177.734 ms/op
```

File: client-v2/src/test/perf/com/clickhouse/client/ClientBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    @Setup
+    public void setup() {
+        BaseIntegrationTest.setupClickHouseContainer();
+
+        ClickHouseNode node = ClickHouseServerForTest.getClickHouseNode(ClickHouseProtocol.HTTP, isCloud(), ClickHouseNode.builder().build());
+        oldClient = new Client.Builder()
+                .addEndpoint(Protocol.HTTP, node.getHost(), node.getPort(), false)
+                .setUsername("default")
+                .setPassword("")
```
Comment (mzitnik) [inline client-v2/src/test/perf/com/clickhouse/client/ClientBenchmark.java:43]: Does our default is empty ("") seems redundant

File: client-v2/src/test/perf/com/clickhouse/client/ClientBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    @Setup
+    public void setup() {
+        BaseIntegrationTest.setupClickHouseContainer();
+
+        ClickHouseNode node = ClickHouseServerForTest.getClickHouseNode(ClickHouseProtocol.HTTP, isCloud(), ClickHouseNode.builder().build());
+        oldClient = new Client.Builder()
+                .addEndpoint(Protocol.HTTP, node.getHost(), node.getPort(), false)
+                .setUsername("default")
```
Comment (mzitnik) [inline client-v2/src/test/perf/com/clickhouse/client/ClientBenchmark.java:42]: it should be our default username also looks redundant

**Review by mzitnik — CHANGES_REQUESTED** [review]
Why we do not have parallel insert test

**Review by mzitnik — APPROVED** [review]

**Pull request merged**